mongo/tests/test_unit.py

```diff
@@ -400,3 +521,74 @@ def test_api_alibaba_mongod(aggregator):
         assert deployment_type.is_arbiter is False
         assert deployment_type.replset_state == 1
         assert deployment_type.replset_name == 'foo'
+
+
+def test_when_replica_check_flag_to_false_then_no_replset_metrics_reported(aggregator, check, instance, dd_run_check):
+    # Given
+    instance['replica_check'] = False
+    check = check(instance)
+    # When
+    with mock_pymongo("replica-primary-in-shard"):
+        dd_run_check(check)
+    # Then
+    reported_metrics = aggregator.metric_names
+    for metric in [
+        'mongodb.replset.health',
+        'mongodb.replset.replicationlag',
+        'mongodb.replset.optime_lag',
+        'mongodb.replset.state',
+        'mongodb.replset.votefraction',
+        'mongodb.replset.votes',
+    ]:
+        assert metric not in reported_metrics
```

Comment:
quick question, is it the same as this?

```python
aggregator.assert_metric(metric, count=0)
```

Reply:
If so, you do not need to get the metric_names yourself

Reply:
Done